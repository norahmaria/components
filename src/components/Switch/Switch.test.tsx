import React from 'react'
import { render } from '@testing-library/react'

import { Switch } from '../index'

describe('Switch Component', () => {
  it('Can render', () => {
    const { getByTestId } = render(
      <Switch label="Switch" id="Switch" onSwitchChange={() => {}} />
    )

    expect(getByTestId('switch-wrapper')).toMatchSnapshot()
  })

  it('Can run the onSwitchChange event on changes', () => {
    const onSwitchChange = jest.fn()

    const { getByTestId } = render(
      <Switch label="Switch" id="Switch" onSwitchChange={onSwitchChange} />
    )

    const label = getByTestId('switch-wrapper')
    label.click()

    expect(onSwitchChange).toHaveBeenCalled()
  })

  it('Does not run onSwitchChange events if disabled', () => {
    const onSwitchChange = jest.fn()

    const { getByTestId } = render(
      <Switch
        label="Switch"
        id="Switch"
        disabled={true}
        onSwitchChange={onSwitchChange}
      />
    )

    const label = getByTestId('switch-wrapper')
    label.click()

    expect(onSwitchChange).not.toHaveBeenCalled()
  })

  it('Can toggle checked / unchecked', () => {
    const { getByTestId } = render(
      <Switch label="Switch" id="Switch" onSwitchChange={() => {}} />
    )

    const label = getByTestId('switch-wrapper')
    label.click()

    const input = getByTestId('switch-input')
    expect(input).toBeChecked()
  })
})
