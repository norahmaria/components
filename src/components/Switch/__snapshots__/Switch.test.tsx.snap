// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch Component Can render 1`] = `
<label
  class="
        switch-nm 
        color-primary 
        size-medium 
        checked-false 
        disabled-false
        undefined
      "
  data-testid="switch-wrapper"
  for="Switch"
  tabindex="1"
>
  <input
    data-testid="switch-input"
    id="Switch"
    name="Switch"
    tabindex="-1"
    type="checkbox"
  />
  <div
    class="switch-nm-toggle"
  >
    <div
      class="dot"
    />
  </div>
  Switch
</label>
`;
